invenio_vocabularies/contrib/affiliations/datastreams.py

```diff
@@ -119,7 +123,114 @@ def _do_update(self, entry):
         return StreamEntry(self._service.update(self._identity, vocab_id, updated))
 
 
-VOCABULARIES_DATASTREAM_READERS = {}
+class EDMOOrganizationHTTPReader(BaseReader):
+    """Reader class to fetch and process EDMO RDF data from the EDMO SPARQL endpoint."""
+
+    def __init__(self, origin=None, mode="r", since=None, *args, **kwargs):
+        """Initialize the reader with the data source.
+
+        :param origin: The SPARQL endpoint from which to fetch the RDF data.
+        :param mode: Mode of operation (default is 'r' for reading).
+        """
+        self._origin = origin or AFFILIATIONS_EDMO_SPARQL_URL
+        super().__init__(origin=origin, mode=mode, *args, **kwargs)
+
+    def _iter(self, fp, *args, **kwargs):
+        raise NotImplementedError(
+            "EDMOOrganizationHTTPReader downloads one result set from SPARQL and therefore does not iterate through items"
+        )
+
+    def read(self, item=None, *args, **kwargs):
+        """Fetch and process the EDMO RDF data, yielding it one organization at a time."""
+        if item:
+            raise NotImplementedError(
+                "EDMOOrganizationHTTPReader does not support being chained after another reader"
+            )
+
+        sparql = SPARQLWrapper(self._origin)
+
+        sparql.setQuery(
+            """
+        SELECT ?org ?name ?altName ?countryName ?locality ?deprecated
+        WHERE {
+            ?org a <http://www.w3.org/ns/org#Organization> .
+            ?org <http://www.w3.org/ns/org#name> ?name .
+            OPTIONAL { ?org <http://www.w3.org/2004/02/skos/core#altName> ?altName } .
+            OPTIONAL { ?org <http://www.w3.org/2006/vcard/ns#country-name> ?countryName } .
+            OPTIONAL { ?org <http://www.w3.org/2006/vcard/ns#locality> ?locality } .
+            ?org <http://www.w3.org/2002/07/owl#deprecated> ?deprecated .
+        }
+        """
+        )
+        sparql.setReturnFormat(JSON)
+
+        results = sparql.query().convert()
+        yield from results["results"]["bindings"]
+
+
+class EDMOOrganizationTransformer(BaseTransformer):
+    """Transformer class to convert EDMO RDF data to a dictionary format."""
+
+    def apply(self, stream_entry, **kwargs):
+        """Applies the transformation to the stream entry."""
+        record = stream_entry.entry
+
+        id_ = record["org"]["value"]
```

Comment:
Here I took the decision of storing the whole URL (e.g. `"https://edmo.seadatanet.org/report/42"`) as the identifier.
Other options could be to store something like `"edmo:42"`.